### PR TITLE
Devel ls development merged

### PR DESCRIPTION
**Pull request by kostasvar**
Pull request for merging the large-scale into the development branch, after resolving conflicts.

### CODE REVIEW

File: code-experiments/src/transform_vars_blockrotation_helpers.c [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+#include <stdio.h>
+#include <assert.h>
+#include "coco.h"
+
+#include "coco_random.c" /*tmp*/
+#include "suite_bbob_legacy_code.c" /*tmp*/
+
+#include <time.h> /*tmp*/
+
+/* TODO: Document this file in doxygen style! */
```
Comment (brockho) [inline code-experiments/src/transform_vars_blockrotation_helpers.c]: @kostasvar: Can we remove this TODO or shall we still take care of it?

File: code-experiments/src/transform_vars_brs.c [hunk 2]
```diff
@@ -43,7 +43,7 @@ static void transform_vars_brs_evaluate(coco_problem_t *problem, const double *x
      * from 1, we use all even indices since C starts indexing
      * with 0.
      */
-    if (x[i] > 0.0 && i % 2 == 0) {
+    if (x[i] > 0.0 && i % 2 == 0) { /*TODO: Documentation: the condition should be Tosz(x_i - x_i^opt) > 0 in stead of z_i > 0*/
```
Comment (brockho) [inline code-experiments/src/transform_vars_brs.c]: @kostasvar: as well here, do we need to keep the TODO?

File: code-experiments/src/transform_vars_permutation_helpers.c [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+#include <stdio.h>
+#include <assert.h>
+#include "coco.h"
+
+#include "coco_random.c"
+#include "suite_bbob_legacy_code.c" /*tmp*/
+
+#include <time.h> /*tmp*/
+
+/* TODO: Document this file in doxygen style! */
```
Comment (brockho) [inline code-experiments/src/transform_vars_permutation_helpers.c]: TODO: check again also here how the doxygen output looks like!

File: code-experiments/src/observer_bbob.c [hunk 4]
```diff
@@ -17,18 +19,43 @@ typedef struct {
...
+
   observer->logger_allocate_function = logger_bbob;
   observer->logger_free_function = logger_bbob_free;
-  observer->data_free_function = NULL;
-  observer->data = NULL;
+  observer->data_free_function = observer_bbob_data_free;
+  observer->data = observer_bbob;
+
 
   *option_keys = NULL;
```
[the remark this answers is not in the document]
Comment (brockho) [reply code-experiments/src/observer_bbob.c]: I am not sure anymore why we have these changes. Running an experiment and postprocessing it with the `observer_bbob.c` from the `master` branch does not show any obvious-to-detect flaw or error.

File: code-experiments/src/transform_vars_brs.c (hunk 2 above)
Comment (brockho) [reply code-experiments/src/transform_vars_brs.c]: Looking at the documentation and at issue #1819 again, we decided that there is no need to update the documentation and therefore, the TODO can be deleted (I will do).

File: code-experiments/src/logger_biobj.c [hunk 5]
```diff
@@ -1,942 +1,942 @@
...
+  }
+
+  coco_free_memory(prefix);
+  coco_free_memory(file_name);
+  coco_free_memory(path_name);
+
+  /* Output header information to the dat file */
+  fprintf(indicator->dat_file, "%%\n%% index = %lu, name = %s\n", (unsigned long) problem->suite_dep_index,
+      problem->problem_name);
+  fprintf(indicator->dat_file, "%% i
```
[the remark this answers is not in the document]
Comment (kostasvar) [reply code-experiments/src/logger_biobj.c]: It should be removed with 
https://github.com/numbbo/coco/pull/1823/commits/b146a09f5a7e9a452ae21289e1337c06a8e8029c

File: code-experiments/src/f_sharp_ridge.c [hunk 6]
```diff
@@ -12,6 +12,10 @@
 #include "transform_obj_shift.c"
 #include "transform_vars_affine.c"
 #include "transform_vars_shift.c"
+#include "transform_vars_conditioning.c"
+#include "transform_vars_permutation.c"
+#include "transform_vars_blockrotation.c"
+#include "transform_obj_norm_by_dim.c"
```
[the remark this answers is not in the document]
Comment (kostasvar) [reply code-experiments/src/f_sharp_ridge.c]: It is fixed with 5936a47a1a5f36833e30bf6fb4e1628cdff43048 .

File: code-experiments/src/old_large_scale_transformations.c [hunk 7]
```diff
@@ -9,8 +9,10 @@
```
[the remark this answers is not in the document]
Comment (kostasvar) [reply code-experiments/src/old_large_scale_transformations.c]: The file is removed with 
https://github.com/numbbo/coco/pull/1823/commits/1669bc3ab4337cfee38f7de4a4e78e43d3d22dbb .

**Pull request merged**